api/envoy/service/ratelimit/v3/rls.proto

```diff
@@ -114,6 +121,14 @@ message RateLimitResponse {
       // Point in time at which the quota expires.
       google.protobuf.Timestamp valid_until = 2;
     }
+
+    // The unique id that is associated with each Quota either at individual descriptor level or whole descriptor set level.
+    //
+    // For the matching policy with disjunction, for example, match: "request.headers['environment'] == 'staging' || request.headers['environment'] == 'dev'"),
+    // the request_headers action produces a distinct list of descriptors for each possible value of the ‘environment’ header even though the granted quota is same.
+    //
+    // The client will use this id information (returned from RLS server) to correctly correlate the multiple descriptors/descriptor sets that have been granted with same quota (i.e., share the same quota among multiple descriptors or descriptor sets.)
```

Comment:
What happens if ID is empty? There is no correlation I assume? Please clarify.

Does the quota refresh when the server returns this if a previous ID was already seen? Please clarify.

What does the client do if the returned quota details are different from the original id? Are they replaced? Ignored? Please clarify.

/wait

Reply:
All are good questions!

> What happens if ID is empty? There is no correlation I assume? Please clarify.
Yes, this is a valid case, there is no correlation and this filed is ignored.

> Does the quota refresh when the server returns this if a previous ID was already seen? Please clarify.

> What does the client do if the returned quota details are different from the original id? Are they replaced? Ignored? Please clarify.

I answered above two questions together here. If newly returned quota (with same id) is different, the old quota will be replaced. 
Refresh will happen in two cases 1) overridden as described above 2) quota expires after `valid_until`